WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Welcome/MigrationWelcomeViewModel.swift

```diff
@@ -0,0 +1,37 @@
+import UIKit
+
+final class MigrationWelcomeViewModel {
+
+    // MARK: - Properties
+
+    var gravatarEmail: String?
+
+    let blogListDataSource: BlogListDataSource
+
+    let configuration: MigrationStepConfiguration
+
+    // MARK: - Init
+
+    init(account: WPAccount?, coordinator: MigrationFlowCoordinator) {
+        if let account {
+            self.gravatarEmail = account.email
+        }
+
+        self.blogListDataSource = BlogListDataSource()
+        self.blogListDataSource.loggedIn = true
+        self.blogListDataSource.account = account
+
+        configuration = MigrationStepConfiguration(headerConfiguration:
+                                                    MigrationHeaderConfiguration(step: .welcome,
+                                                                                 multiSite: blogListDataSource.visibleBlogsCount > 1),
+                                                   actionsConfiguration:
+                                                    MigrationActionsViewConfiguration(step: .welcome,
+                                                                                      primaryHandler: { [weak coordinator] in
+            coordinator?.transitionToNextStep()
+        },
+                                                                                      secondaryHandler: {
+
+        }))
```

Comment:
This is just a nice-to-have, but I honestly find this code a bit hard to read 🙈

Maybe we should breakdown this complex expression into multiple variables, something like this:

```swift
let headerConfig = MigrationHeaderConfiguration(step: .welcome, multiSite: blogListDataSource.visibleBlogsCount > 1)
let actionsConfig = MigrationActionsViewConfiguration(step: .welcome, primaryHandler:...)
configuration = MigrationStepConfiguration(headerConfiguration: headerConfig, actionsConfig: actionsConfig)

```

Reply:
Totally agree. I did this in all view models